packages/adaptivecpp/package.py

```diff
@@ -0,0 +1,252 @@
+            )
+            if len(nvcpp_cands) < 1:
+                raise InstallError("Failed to find nvc++ executable")
+            args.append("-DNVCXX_COMPILER={0}".format(nvcpp_cands[0]))
+
+            if not ("llvm" in spec):
+                args.append("-DWITH_CUDA_NVCXX_ONLY=ON")
+
+        if not ("llvm" in spec):
+            args += [
+                "-DWITH_ACCELERATED_CPU=OFF",
+                "-DBUILD_CLANG_PLUGIN=OFF",
+            ]
+
+        if "+opencl" in spec:
+            args += [
+                "-DWITH_SSCP_COMPILER:Bool=TRUE",
+                "-DWITH_OPENCL_BACKEND=ON",
+            ]
+        else:
+            args += [
+                "-DWITH_SSCP_COMPILER:Bool=FALSE",
+                "-DWITH_OPENCL_BACKEND=OFF",
+            ]
+
+        return args
+
+    @run_after("install")
+    def filter_config_file(self):
+
+        # The config file name and location depends on version:
+        # pre-24.02.0: syclcc.json
+        # post-24.02.0: acpp-core.json
+        config_file_paths = filesystem.find(
+            self.prefix, ("syclcc.json", "acpp-core.json")
+        )
+        if len(config_file_paths) != 1:
+            raise InstallError(
+                "installed AdaptiveCPP must provide a unique compiler driver "
+                "configuration file, found: {0}".format(config_file_paths)
+            )
+        config_file_path = config_file_paths[0]
+        with open(config_file_path) as f:
+            config = json.load(f)
+        # 1. Fix compiler: use the real one in place of the Spack wrapper
+        config["default-cpu-cxx"] = self.compiler.cxx
+        # 2. Fix stdlib: we need to make sure cuda-enabled binaries find
+        #    the libc++.so and libc++abi.so dyn linked to the sycl
+        #    ptx backend
+
+        if "llvm" in self.spec:
+            rpaths = set()
+            so_paths = filesystem.find(self.spec["llvm"].prefix, "libc++.so")
+            if len(so_paths) != 1:
+                raise InstallError(
+                    "concretized llvm dependency must provide a "
+                    "unique directory containing libc++.so, "
+                    "found: {0}".format(so_paths)
+                )
+            rpaths.add(path.dirname(so_paths[0]))
+            so_paths = filesystem.find(self.spec["llvm"].prefix, "libc++abi.so")
+            if len(so_paths) != 1:
+                raise InstallError(
+                    "concretized llvm dependency must provide a "
+                    "unique directory containing libc++abi.so, "
+                    "found: {0}".format(so_paths)
+                )
+            rpaths.add(path.dirname(so_paths[0]))
+            default_cuda_link_line = "default-cuda-link-line"
+            if default_cuda_link_line in config.keys():
+                config[default_cuda_link_line] += " " + " ".join(
+                    "-rpath {0}".format(p) for p in rpaths
+                )
```

Comment:
Should you also deal with the case where this key is not already present?

Reply:
I think if they key is not present then nothing needs to be done. The hipsycl package (which was the source) assumes that the key is always present and this turned out not to be true in later versions.

Reply:
Well, if everything works then I guess it's fine as it is.